src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -584,6 +584,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 if (end_constant_node == nullptr)
                     return false;
                 auto end_values = end_constant_node->cast_vector<int64_t>();
+                if (end_values[concat_axis] == std::numeric_limits<int64_t>::max())
```

Comment:
it believe it's better to check that `end_values[concat_axis] > concat->get_shape()[concat_axis]` , `end_values[concat_axis] == std::numeric_limits<int64_t>::max()` is a corner case

according to the strided slice specification
<img width="465" alt="{5E2F3BD7-A6F2-4A53-AEB4-7EE5D4282D72}" src="https://github.com/user-attachments/assets/5fe51473-a193-4937-a8ce-e0d4b108fc99" />

Reply:
Good idea